packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -561,6 +564,14 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
       product.servingSize = changedProduct.servingSize;
       product.nutriments = changedProduct.nutriments;
       await daoProduct.put(product);
+      provider.set(product);
+    } else {
+      final Product newProduct = Product(
+        servingSize: changedProduct.servingSize,
+        nutriments: changedProduct.nutriments,
+      );
```

Comment:
Of course, the same process as in `SimpleInputPage` must be written there.
Besides, you forgot to populate the `barcode`, which is evil ;)

Reply:
Update, made the changes to `simpleInputPage `and `editIngredient `page
For the rest two pages , it was proving hard to implement something similar to this (Getting stuck in that), Besides the current method works. 
Maybe if you could have a look again, and see if it's good to go with.

Thanks